### PR TITLE
view list cart

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
xem danh sách sản phẩm trong cửa hàng
[- ticket redmine](https://edu-redmine.sun-asterisk.vn/issues/74322)

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)
image
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/service/cart.service.ts [hunk 1]
```diff
@@ -56,4 +60,66 @@ export class CartsService {
...
+        // calculation price all toppings
+        const totalToppingPrice = toppings.reduce((acc, topping) => {
+          return acc + topping.price;
+        }, 0);
+        // calculation total product includes
+        //(price product, price toppings, price up_size)
+        const totalProduct =
+          (item?.productInstance?.product?.price + totalToppingPrice) *
+            item?.quantity +
+          (item?.up_size ? 5000 : 0);
```
Comment (huongnt-2545) [inline src/service/cart.service.ts]: `5000` -> khai báo trong file constants e nha

File: src/service/product.service.ts [hunk 2]
```diff
@@ -54,6 +53,13 @@ export class ProductsService {
     }
   }
 
+  public getToppingByProduct(productInstanceId: number) {
```
Comment (huongnt-2545) [inline src/service/product.service.ts]:
```suggestion
  public getToppingByProduct(productIds: number) {
```

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**